cpp/src/binaryop/jit/code/operation.cpp

```diff
@@ -40,16 +40,34 @@ R"***(
     using RAdd = Add;
 
     struct Sub {
-        template <typename TypeOut, typename TypeLhs, typename TypeRhs>
-        static TypeOut operate(TypeLhs x, TypeRhs y) {
+        template <typename TypeOut, typename TypeLhs, typename TypeRhs,
+                  enable_if_t<(!is_integral_v<TypeOut> ||
+                               !is_timestamp_t_v<TypeLhs> ||
+                               !is_timestamp_t_v<TypeRhs>)>* = nullptr>
+        static TypeOut operate(TypeLhs x, TypeRhs y) {
+            // Note: difference of time_point's should be convertible to TypeOut. The period
```

Comment:
Are these comments valid in this version too?

Reply:
yes. it simply states that the difference of 2 time_points should be supported and that the result (larger duration precision amongst the two) should be convertible to `Typeout`. simply put, `Typeout` should have a duration precision >= the precision from the difference - basically, what `std::chrono` allows today.

Reply:
I mean I see this comment in two places. This seems like it should only be in the other sfinae overload. I’m pointing to the overload that is for non time stamp types.

Reply:
this method will be specialized if both the operands *and* the result type *all* happens to be timestamps. hence, this is valid here. it can be removed from the other place.